code/lib/builder-vite/src/vite-config.ts

```diff
@@ -91,7 +91,7 @@ export async function pluginConfig(options: ExtendedOptions) {
   ] as PluginOption[];
 
   // We need the react plugin here to support MDX in non-react projects.
-  if (frameworkName !== '@storybook/react-vite') {
+  if (frameworkName === '@storybook/react-vite') {
```

Comment:
Thanks for the PR, but unfortunately this isn't what we want to do here.  We're using the `@vitejs/plugin-react` plugin to handle MDX, so we need to be sure it's installed.

I think maybe the issue is that I've assumed that folks using the react storybook framework will also be using the react vite plugin, but that may not always be the case.  

So, could you share a few pieces of information to help us troubleshoot / confirm what's happening?

- What version of storybook are you using?
- What is in your `vite.config.js` and `.storybook/main.js`?

Thanks!

Reply:
storybook version: `7.0.0-alpha.33`
Here is my `.storybook/main.js` config, which is very simple：
```js
module.exports = {
  stories: ['../stories/**/*.(mdx|stories.jsx)'],
  addons: [
    '@storybook/addon-links',
    '@storybook/addon-essentials',
    '@storybook/addon-interactions',
  ],
  framework: {
    name: '@storybook/react-vite',
  },
}
```
I just changed framework from `@storybook/react-webpack5` to `@storybook/react-vite`.
And there is no `vite.config.js` file in my project, using the default configuration of `@storybook/react-vite`.

Therefore, when `frameworkName = '@storybook/react-vite'`, `@vitejs/plugin-react` will not be push into plugins:

```ts
// code/lib/builder-vite/src/vite-config.ts
if (frameworkName !== '@storybook/react-vite') {
  plugins.push(viteReact({ exclude: [/\.stories\.([tj])sx?$/, /node_modules/, /\.([tj])sx?$/] }));
}
```

This will cause `mdxPlugin` to get no `reactRefreshPlugins` and report an error:

```ts
// code/lib/builder-vite/src/plugins/mdx-plugin.ts
{
  name: 'storybook:mdx-plugin',
  enforce: 'pre',
  configResolved({ plugins }) {
    // @vitejs/plugin-react-refresh has been upgraded to @vitejs/plugin-react,
    // and the name of the plugin performing `transform` has been changed from 'react-refresh' to 'vite:react-babel',
    // to be compatible, we need to look for both plugin name.
    // We should also look for the other plugins names exported from @vitejs/plugin-react in case there are some internal refactors.
    const reactRefreshPlugins = plugins.filter(
      (p) =>
        p.name === 'react-refresh' ||
        p.name === 'vite:react-babel' ||
        p.name === 'vite:react-refresh' ||
        p.name === 'vite:react-jsx'
    );
    reactRefresh = reactRefreshPlugins.find((p) => p.transform);
  },
}
```

Reply:
OK, I have an idea of how to address this.  I'll push up a PR soon.